compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala

```diff
@@ -94,8 +94,8 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
     lazy val accessors =
       if (isDerivedValueClass(clazz)) clazz.paramAccessors.take(1) // Tail parameters can only be `erased`
       else clazz.caseAccessors
-    val isEnumCase = clazz.derivesFrom(defn.EnumClass) && clazz != defn.EnumClass
-    val isEnumValue = isEnumCase && clazz.isAnonymousClass && clazz.classParents.head.classSymbol.is(Enum)
+    val isEnumCase = clazz.isEnum || clazz.classParents.exists(_.classSymbol.isEnum)
```

Comment:
Would that not make an enum class itself an enum case? So the name `enumCase` seems to be misleading, and I am not sure the logic is correct.

Reply:
In fact, it seems to be better to just drop `isEnumCase` since it is only used in `isEnumValue`, and define `isEnumValue` like this:
```scala
val isEnumValue = clazz.isAnonymousClass && clazz.classParents.head.classSymbol.is(Enum)
```

Reply:
added this change